codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/customize/SerdeDecorator.kt

```diff
@@ -0,0 +1,28 @@
+/*
+* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+* SPDX-License-Identifier: Apache-2.0
+*/
+
+package software.amazon.smithy.rust.codegen.client.smithy.customize
+
+import software.amazon.smithy.rust.codegen.client.smithy.ClientCodegenContext
+import software.amazon.smithy.rust.codegen.core.rustlang.Feature
+import software.amazon.smithy.rust.codegen.core.smithy.RustCrate
+
+/**
+ * This class,
+ * - Adds serde as a dependency
+ *
+ */
+class SerdeDecorator : ClientCodegenDecorator {
+    override val name: String = "SerdeDecorator"
+    override val order: Byte = -1
+
+    override fun extras(codegenContext: ClientCodegenContext, rustCrate: RustCrate) {
```

Comment:
Would be good to also hook into the `ModuleDocSection.ServiceDocs`, or even add a new `ModuleDocSection.Features`, so that these features can be described in the docs. That way, the documentation can inform on the `--cfg aws_sdk_unstable` requirement. This would require registering a `LibRsCustomization`, so `override fun libRsCustomizations(...)`.

There's an example here that adds information to the "Crate Organization" documentation section: https://github.com/awslabs/smithy-rs/blob/2f60a5e0904037ec5237c080c61817617b9d8c06/codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/customizations/ClientCustomizations.kt#L19

Reply:
I added this function but it seems like it's the generated code is not affected.

Is there anything that I should be doing?